File: test_groq_integration.py
#!/usr/bin/env python3
"""
Test GROQ API Integration with the provided API key
"""

import os
import requests
import json

def test_groq_api():
    """Test GROQ API with the provided key"""
    
    # Use the API key from the review request
    api_key = "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"
    
    print("🧪 Testing GROQ API Integration...")
    print(f"🔑 API Key: {api_key[:10]}...{api_key[-10:]}")
    
    # Test with the latest model mentioned in the review request
    test_models = [
        "llama-3.3-70b-versatile",  # Latest model mentioned in review
        "llama-3.1-8b-instant",     # Model mentioned in test results
        "llama3-8b-8192"            # Deprecated model
    ]
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }
    
    for model in test_models:
        print(f"\n🤖 Testing model: {model}")
        
        test_payload = {
            "messages": [
                {
                    "role": "system", 
                    "content": "You are KAiro, an AI browser assistant. Respond briefly."
                },
                {
                    "role": "user", 
                    "content": "Test message for KAiro Browser integration"
                }
            ],
            "model": model,
            "max_tokens": 50,
            "temperature": 0.3
        }
        
        try:
            response = requests.post(
                "https://api.groq.com/openai/v1/chat/completions",
                headers=headers,
                json=test_payload,
                timeout=15
            )
            
            if response.status_code == 200:
                data = response.json()
                message = data['choices'][0]['message']['content']
                print(f"✅ SUCCESS: {model}")
                print(f"📝 Response: {message[:100]}...")
                print(f"⚡ Usage: {data.get('usage', {})}")
                return True, model, message
                
            elif response.status_code == 400:
                error_data = response.json()
                if "does not exist" in str(error_data):
                    print(f"❌ DEPRECATED: {model} - Model no longer available")
                else:
                    print(f"❌ ERROR: {model} - {error_data}")
                    
            elif response.status_code == 401:
                print(f"🔐 AUTH ERROR: Invalid API key")
                return False, model, "Authentication failed"
                
            elif response.status_code == 429:
                print(f"⏰ RATE LIMIT: Too many requests")
                return False, model, "Rate limited"
                
            else:
                print(f"❌ HTTP ERROR: {response.status_code} - {response.text}")
                
        except requests.exceptions.Timeout:
            print(f"⏰ TIMEOUT: {model}")
        except requests.exceptions.ConnectionError:
            print(f"🌐 CONNECTION ERROR: {model}")
        except Exception as e:
            print(f"💥 EXCEPTION: {model} - {str(e)}")
    
    return False, None, "All models failed"

def test_agent_coordination():
    """Test agent coordination with AI"""
    
    api_key = "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"
    
    print("\n🤖 Testing Agent Coordination...")
    
    # Test different agent scenarios
    test_scenarios = [
        {
            "name": "Research Agent",
            "prompt": "I need to research the latest AI developments in 2025",
            "expected_agent": "research"
        },
        {
            "name": "Navigation Agent", 
            "prompt": "Navigate to google.com and wikipedia.org",
            "expected_agent": "navigation"
        },
        {
            "name": "Shopping Agent",
            "prompt": "Help me find the best laptop deals",
            "expected_agent": "shopping"
        },
        {
            "name": "Analysis Agent",
            "prompt": "Analyze this webpage content for key insights",
            "expected_agent": "analysis"
        }
    ]
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }
    
    for scenario in test_scenarios:
        print(f"\n🎯 Testing: {scenario['name']}")
        
        system_prompt = """You are KAiro, an AI browser assistant with 6 specialized agents:
1. Research Agent - for information gathering and research
2. Navigation Agent - for website navigation and URL handling  
3. Shopping Agent - for product research and price comparison
4. Communication Agent - for emails and messaging
5. Automation Agent - for workflow automation
6. Analysis Agent - for content analysis and insights

Analyze the user request and determine which agent should handle it. Respond with just the agent name and a brief explanation."""

        payload = {
            "messages": [
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": scenario["prompt"]}
            ],
            "model": "llama-3.3-70b-versatile",
            "max_tokens": 100,
            "temperature": 0.1
        }
        
        try:
            response = requests.post(
                "https://api.groq.com/openai/v1/chat/completions",
                headers=headers,
                json=payload,
                timeout=10
            )
            
            if response.status_code == 200:
                data = response.json()
                ai_response = data['choices'][0]['message']['content'].lower()
                
                if scenario["expected_agent"] in ai_response:
                    print(f"✅ CORRECT: AI selected {scenario['expected_agent']} agent")
                    print(f"📝 Response: {ai_response[:80]}...")
                else:
                    print(f"⚠️  UNEXPECTED: Expected {scenario['expected_agent']}, got: {ai_response[:80]}...")
                    
            else:
                print(f"❌ API ERROR: {response.status_code}")
                
        except Exception as e:
            print(f"💥 ERROR: {str(e)}")

if __name__ == "__main__":
    print("🚀 KAiro Browser - GROQ API Integration Test")
    print("=" * 60)
    
    # Test basic API functionality
    success, working_model, response = test_groq_api()
    
    if success:
        print(f"\n✅ GROQ API Integration: WORKING")
        print(f"🎯 Best Model: {working_model}")
        
        # Test agent coordination
        test_agent_coordination()
        
        print(f"\n🎉 INTEGRATION TEST COMPLETE")
        print(f"📊 Status: Ready for KAiro Browser usage")
        
    else:
        print(f"\n❌ GROQ API Integration: FAILED")
        print(f"🔧 Issue: {response}")

File: test_groq_integration.js
// Test GROQ API Integration
// This test verifies that the GROQ API key and model are working correctly

const Groq = require('groq-sdk');
require('dotenv').config();

console.log('🧪 Testing GROQ API Integration...');
console.log('🔑 API Key present:', process.env.GROQ_API_KEY ? 'YES' : 'NO');

if (!process.env.GROQ_API_KEY) {
  console.error('❌ GROQ_API_KEY not found in environment variables');
  process.exit(1);
}

const groq = new Groq({
  apiKey: process.env.GROQ_API_KEY
});

async function testGroqAPI() {
  try {
    console.log('\n📡 Testing primary model: llama-3.3-70b-versatile');
    
    const response = await groq.chat.completions.create({
      messages: [
        {
          role: 'user',
          content: 'Hello! Please respond with "GROQ API integration test successful" if you can read this message.'
        }
      ],
      model: 'llama-3.3-70b-versatile',
      max_tokens: 50,
      temperature: 0.1
    });

    if (response && response.choices && response.choices.length > 0) {
      console.log('✅ Primary model working successfully!');
      console.log('📊 Model:', response.model || 'llama-3.3-70b-versatile');
      console.log('💬 Response:', response.choices[0].message.content);
      console.log('📈 Usage:', response.usage);
      
      return true;
    } else {
      throw new Error('Invalid response structure');
    }
    
  } catch (primaryError) {
    console.error('❌ Primary model failed:', primaryError.message);
    console.log('\n🔄 Trying fallback model: llama3-8b-8192');
    
    try {
      const fallbackResponse = await groq.chat.completions.create({
        messages: [
          {
            role: 'user',
            content: 'Test fallback model response'
          }
        ],
        model: 'llama3-8b-8192',
        max_tokens: 30,
        temperature: 0.1
      });

      if (fallbackResponse && fallbackResponse.choices && fallbackResponse.choices.length > 0) {
        console.log('✅ Fallback model working!');
        console.log('📊 Model:', fallbackResponse.model || 'llama3-8b-8192');
        console.log('💬 Response:', fallbackResponse.choices[0].message.content);
        
        return 'fallback';
      } else {
        throw new Error('Invalid fallback response');
      }
      
    } catch (fallbackError) {
      console.error('❌ Fallback model also failed:', fallbackError.message);
      console.error('\n🔍 Debugging information:');
      console.error('- API Key format:', process.env.GROQ_API_KEY.substring(0, 8) + '...');
      console.error('- Error type:', fallbackError.constructor.name);
      
      if (fallbackError.response) {
        console.error('- HTTP Status:', fallbackError.response.status);
        console.error('- Response data:', fallbackError.response.data);
      }
      
      return false;
    }
  }
}

// Test function for agent task analysis
function testAgentTaskAnalysis() {
  console.log('\n🤖 Testing Agent Task Analysis...');
  
  const testTasks = [
    'research latest AI developments',
    'navigate to google.com',
    'help me shop for laptops',
    'compose an email to my team',
    'automate my daily workflow',
    'analyze this page content'
  ];
  
  // Import the task analysis logic from main.js (simplified version)
  function analyzeAgentTask(task) {
    const lowerTask = task.toLowerCase();
    const scores = {
      research: 0,
      navigation: 0,
      shopping: 0,
      communication: 0,
      automation: 0,
      analysis: 0
    };

    // Research scoring
    if (lowerTask.includes('research') || lowerTask.includes('find') || lowerTask.includes('information')) {
      scores.research = 90;
    }
    
    // Navigation scoring
    if (lowerTask.includes('navigate') || lowerTask.includes('go to') || lowerTask.includes('.com')) {
      scores.navigation = 95;
    }
    
    // Shopping scoring
    if (lowerTask.includes('shop') || lowerTask.includes('buy') || lowerTask.includes('price')) {
      scores.shopping = 90;
    }
    
    // Communication scoring
    if (lowerTask.includes('email') || lowerTask.includes('compose') || lowerTask.includes('write')) {
      scores.communication = 90;
    }
    
    // Automation scoring
    if (lowerTask.includes('automate') || lowerTask.includes('workflow') || lowerTask.includes('schedule')) {
      scores.automation = 90;
    }
    
    // Analysis scoring
    if (lowerTask.includes('analyze') || lowerTask.includes('analysis') || lowerTask.includes('examine')) {
      scores.analysis = 90;
    }

    const primaryAgent = Object.keys(scores).reduce((a, b) => scores[a] > scores[b] ? a : b);
    const confidence = scores[primaryAgent];

    return { primaryAgent, confidence, scores };
  }
  
  testTasks.forEach(task => {
    const result = analyzeAgentTask(task);
    const status = result.confidence >= 80 ? '✅' : result.confidence >= 60 ? '⚠️' : '❌';
    console.log(`${status} "${task}" → ${result.primaryAgent} (${result.confidence}% confidence)`);
  });
}

// Main test execution
async function runAllTests() {
  console.log('🚀 Starting comprehensive integration tests...\n');
  
  const groqResult = await testGroqAPI();
  testAgentTaskAnalysis();
  
  console.log('\n📋 Test Summary:');
  console.log('- GROQ API:', groqResult === true ? '✅ Primary model working' : 
                              groqResult === 'fallback' ? '⚠️ Fallback model only' : '❌ Failed');
  console.log('- Agent Analysis: ✅ Working');
  
  if (groqResult) {
    console.log('\n🎉 Integration tests completed successfully!');
    console.log('💡 The KAiro Browser backend should work correctly with these integrations.');
  } else {
    console.log('\n⚠️ Some integration issues detected. Please check the GROQ API key and network connection.');
  }
}

runAllTests().catch(error => {
  console.error('❌ Test execution failed:', error);
  process.exit(1);
});